### DISCUSSION TITLE
Possible invalid versioning causing update PRs to not be created for shieldsio/shields docker image

### DISCUSSION THREAD

**Discussion by lluiscab** [renovatebot/renovate#28218] [Request Help] [answered]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab

### Please tell us more about your question or problem

I have a repository containing a `docker-compose.yml` file, this file has the following content:

```yaml
version: '3.7'

services:
  shields:
    container_name: shields
    image: shieldsio/shields:server-2023-03-01
    restart: always
    (...)
```

Even tho there are newer versions of the `shieldsio/shields` docker image (latest is `shieldsio/shields:server-2024-04-01` from Apr 1), renovate has never created a single PR for it.

I've attached a section of the logs, in which it can be seen that `shieldsio/shields` is ignored due to `unsupported/unversioned value server-2023-03-01 (versioning=docker)`

I'd say that this versioning schema (`server-YYYY-MM-DD`) is not at all common and there's no implementation for handling it and comparing it to check for newer versions.

Is this something that can be fixed adding a `packageRules` for `shieldsio/shields` with some special configuration for versioning or is this going to need a new versioning scheme to be implemented?

### Logs (if relevant)

<details><summary>Logs</summary>

```
(...)
DEBUG: Matched 1 file(s) for manager docker-compose: docker-compose.yml (repository=****)
DEBUG: docker-compose.extractPackageFile(docker-compose.yml) (repository=****)
DEBUG: manager extract durations (ms) (repository=****)
       "managers": {"docker-compose": 4}
DEBUG: Found docker-compose package files (repository=****)
DEBUG: Found 1 package file(s) (repository=****)
 INFO: Dependency extraction complete (repository=****, baseBranch=main)
       "stats": {
         "managers": {"docker-compose": {"fileCount": 1, "depCount": 1}},
         "total": {"fileCount": 1, "depCount": 1}
       }
DEBUG: Dependency shieldsio/shields has unsupported/unversioned value server-2023-03-01 (versioning=docker) (repository=****)
DEBUG: Skipping shieldsio/shields because no currentDigest or pinDigests (repository=****)
DEBUG: PackageFiles.add() - Package file saved for base branch (repository=****, baseBranch=main)
DEBUG: Package releases lookups complete (repository=****, baseBranch=main)
DEBUG: branchifyUpgrades (repository=****)
DEBUG: detectSemanticCommits() (repository=****)
DEBUG: getCommitMessages (repository=****)
DEBUG: semanticCommits: detected "angular" (repository=****)
DEBUG: semanticCommits: enabled (repository=****)
DEBUG: 0 flattened updates found:  (repository=****)
DEBUG: Returning 0 branch(es) (repository=****)
DEBUG: config.repoIsOnboarded=true (repository=****)
DEBUG: packageFiles with updates (repository=****, baseBranch=main)
       "config": {
         "docker-compose": [
           {
             "deps": [
               {
                 "depName": "shieldsio/shields",
                 "currentValue": "server-2023-03-01",
                 "replaceString": "shieldsio/shields:server-2023-03-01",
                 "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
                 "datasource": "docker",
                 "updates": [],
                 "packageName": "shieldsio/shields",
                 "versioning": "docker",
                 "warnings": [],
                 "skipReason": "invalid-value"
               }
             ],
             "packageFile": "docker-compose.yml"
           }
         ]
       }
(...)
```
</details>

**Comment by rarkins** [comment]
Not a bug, as Renovate can't promise to guess every versioning scheme (although we are happy to add rules to detect/improve).

At least a couple of ways to solve your case using config including:
- versionCompatibility + loose versioning
- regex versioning

**Comment by lluiscab** [comment, marked as answer]
regex versioning works perfectly. not sure how I missed it! Thank you very much.

Leaving the new configuration here for future reference in case anyone else has a similar issue:
```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "packageRules": [
    {
      "matchPackageNames": [
        "shieldsio/shields"
      ],
      "versioning": "regex:^(?<compatibility>.*)-(?<major>\\d+)-(?<minor>\\d+)-(?<patch>\\d+)$"
    }
  ]
}
```
